presto-orc/src/main/java/com/facebook/presto/orc/AbstractOrcRecordReader.java

```diff
@@ -532,8 +539,14 @@ private void advanceToNextStripe()
 
         StripeInformation stripeInformation = stripes.get(currentStripe);
         validateWriteStripe(stripeInformation.getNumberOfRows());
+        List<Slice> stripeDecryptionKeyMetadata = getDecryptionKeyMetadata(currentStripe);
+        if ((!stripeDecryptionKeyMetadata.isEmpty() && !currentGroupDwrfDecryptors.isPresent())
+                || (currentGroupDwrfDecryptors.isPresent() && stripeDecryptionKeyMetadata.equals(currentGroupDwrfDecryptors.get().getKeyMetadatas()))) {
+            verify(decryptorProvider.isPresent(), "decryptorProvider is absent");
+            currentGroupDwrfDecryptors.equals(new GroupDwrfDecryptors(decryptorProvider.get(), stripeDecryptionKeyMetadata));
```

Comment:
`currentGroupDwrfDecryptors ` is `Optional`. This condition is always false.

Reply:
`currentGroupDwrfDecryptors ` is never set

Reply:
ah sorry, that was supposed to be setting the value.